core/src/main/scala/org/apache/spark/deploy/history/HistoryPage.scala

```diff
@@ -34,7 +34,7 @@ private[history] class HistoryPage(parent: HistoryServer) extends WebUIPage("")
     val requestedIncomplete =
       Option(request.getParameter("showIncomplete")).getOrElse("false").toBoolean
 
-    val allApps = parent.getApplicationList().filter(_.completed != requestedIncomplete)
+    val allApps = parent.getApplicationList(true).filter(_.completed != requestedIncomplete)
```

Comment:
Same comment. Forcing a refresh of the listing is expensive, and doesn't sound related to this PR.